Readme.md

```diff
@@ -1,21 +1,146 @@
+- [Contract-ethers](https://github.com/aragon/osx/tree/develop/packages/contracts-ethers): containing the connection between the ethers package and our contracts.
+
+For more information on the individual packages, please read the respective `README.md`.
+
+## Contributing
+
+We'd love to hear what you think! If you want to build this with us, please find a detailed contribution guide in the `CONTRIBUTION_GUIDE.md` [file here](https://github.com/aragon/osx/blob/develop/CONTRIBUTION_GUIDE.md).
 
 ## Setup
 
-Start by running `yarn install` in the root.
+Start by running `yarn install` in the root in your terminal.
 
 ### Dependencies
 
 Since the repo is set up as yarn workspace, all the linking is done automatically.
 
+## How the Aragon OSx protocol works
+
+To review the contracts powering the Aragon OSx protocol, feel free to head to `packages/contracts`.
+
+The Aragon OSx protocol architecture is composed of two key sections:
+
+- __Core contracts__: details the functionalities all DAOs will have. It is composed of mostly 3 sections:
+    - **DAO treasury:** the DAO’s main contract and vault.
+    - **Permission manager**: manages permissions within the DAO to execute actions.
+    - **Plugin manager**: manages installed plugins within the DAO.
+- __Framework contracts__: in charge of creating and registering each deployed DAO or plugin. It contains:
+    - **DAO and Plugin Factories**: creates DAOs or plugins.
+    - **DAO and Plugin Registries**: registers into our protocol those DAOs or plugins.
+    - **Plugin Processor:** installs and uninstalls plugins into DAOs.
+
+Additionally to those two sections, we have developed several plugins DAOs can easily install upon creation. These are:
+
+- __Token Voting plugin__: enabling voting through token majority as a governance mechanism for DAOs
+- __Multisig plugin__: enabling assets to be transacted based on approval from members.
+- __Admin plugin__: enabling one account to have god-like access to the DAO.
+- __Addresslist Voting plugin__: a set of pre-selected addresses are able to vote within a DAO.
+
+Let's dive into more detail on each of these sections.
+
+### Core Contracts
+
+The *Core Contracts* describe how every DAO generated in the Aragon OSx protocol will be set up. It is extremely lean and by design and constitutes the most critical aspects of our architecture.
+
+In a nutshell, each DAO is composed of 3 interconnecting components:
+
+1. **The DAO treasury:** The DAO contract is where the **core functionality** of the DAO lies. It is the contract in charge of:
+    - Representing the identity of the DAO (ENS name, logo, description, other metadata)
+    - Holding and managing the treasury
+    - Executing arbitrary actions to:
+        - transfer assets
+        - call its own functions
+        - call functions in external contracts
+    - Providing general technical utilities (signature validation, callback handling)
```

Comment:
signature validation is handled by the `signatureValidator` that can be set in the DAO but not the DAO itself